openage/util/profiler.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2015-2017 the openage authors. See copying.md for legal info.
+
+"""
+A Profiling Utility.
+"""
+
+import cProfile
+import pstats
+import io
+
+
+class Profiler:
+    """A class for quick and easy profiling.
+        Usage:
+            p = Profiler()
+            with p:
+                #Call methods that need to be profiled.
+            p.report()
+    """
+
+    profile = None
+    profile_stats = None
+    profile_stream = None
+
+    def __init__(self, oStream=None):
+        # oStream can be a file if the profile results want to be saved.
+        self.profile = cProfile.Profile()
+        self.profile_stream = oStream
+
+    def __enter__(self):
+        # begins profiling
+        self.profile.clear()
+        self.profile.enable()
```

Comment:
I think it would be cool if the enable and disable functions are not only exposed by the context manager (`with`) but also with `.enable()` and `.disable()` methods provided by the `Profiler` class. That way one could, sometimes, avoid indenting larger blocks. What do you think?